Parser/Client/Browser.php

```diff
@@ -1119,6 +1149,14 @@ public function parse(): ?array
             $family = 'Firefox';
         }
 
+        /**
+         * Mobile Safari version is always the iOS / iPadOS version
+         * See https://developer.apple.com/documentation/safari-release-notes
+         */
+        if (\in_array($this->osName, ['iOS', 'iPadOS']) && '' !== $this->osVersion && 'Mobile Safari' === $name) {
+            $version = $this->osVersion;
+        }
+
```

Comment:
Maybe just this condition will be enough for us?
```php
        // Mobile Safari version is always the iOS / iPadOS version
        // See https://developer.apple.com/documentation/safari-release-notes
        if ('Mobile Safari' === $name && '' === $version) {
            \preg_match(
                '~iP(?:hone|ad) ?OS ([0-9]{1,2})[_.]([0-9]{1,2})(?:[_.]([0-9]{1,2}))?~i',
                $this->userAgent,
                $osMatch
            );
            if ($osMatch) {
                $version = \implode('.', \array_filter([$osMatch[1], $osMatch[2], $osMatch[3] ?? null]));
            }
        }
        
```

Then our browser parser will not depend on  OS parser.
This will also work correctly when using BrowserParser alone.

Reply:
> Maybe just this condition will be enough for us?
> 
> Then our browser parser will not depend on OS parser. This will also work correctly when using BrowserParser alone.

It will not work because there are user agents like:

`MobileSafari/9537.53 CFNetwork/672.1.13 Darwin/13.1.0`